core/src/Streamly/Internal/Data/Array.hs

```diff
@@ -560,13 +560,10 @@ pinnedSerialize = encodeAs Pinned
 -- | Decode a Haskell type from a byte array containing its serialized
 -- representation.
 {-# INLINE deserialize #-}
-deserialize :: Serialize a => Array Word8 -> a
-deserialize arr@(Array {..}) = unsafeInlineIO $ do
-    let lenArr = length arr
-    (off, val) <-
-        Serialize.deserializeAt arrStart arrContents (arrStart + lenArr)
-    assertM(off == arrStart + lenArr)
-    pure val
+deserialize :: Serialize a => Array Word8 -> (a, Array Word8)
+deserialize arr =
+    let (a, b) = unsafeInlineIO $ MA.deserialize (unsafeThaw arr)
+     in (a, unsafeFreeze b)
```

Comment:
can just use `fmap unsafeFreeze`.

Reply:
I don't like using `fmap` on tuples. To me, tuple as a functor is not very intuitive.